src/support.cpp

```diff
@@ -1399,12 +1372,24 @@ void AreaSupport::handleTowers(const Settings& settings, Polygons& supportLayer_
     // make tower roofs
     const coord_t layer_thickness = settings.get<coord_t>("layer_height");
     const AngleRadians tower_roof_angle = settings.get<AngleRadians>("support_tower_roof_angle");
-    const double tan_tower_roof_angle = tan(tower_roof_angle);
-    const coord_t tower_roof_expansion_distance = layer_thickness / tan_tower_roof_angle;
     const coord_t tower_diameter = settings.get<coord_t>("support_tower_diameter");
+    const coord_t support_line_width = settings.get<coord_t>("support_line_width");
+
+    coord_t tower_roof_expansion_distance;
+    if (tower_roof_angle == 0)
+    {
+        // maximum expansion distance needed to reach a tower_diameter^2 tower area
+        tower_roof_expansion_distance = tower_diameter / 2;
+    }
+    else
+    {
+        const double tan_tower_roof_angle = tan(tower_roof_angle);
+        tower_roof_expansion_distance = layer_thickness / tan_tower_roof_angle;
+    }
+
     for (Polygons& tower_roof: tower_roofs)
     {
-        if (tower_roof.size() == 0)
+        if (tower_roof.empty())
         {
             continue;
         }
```

Comment:
consider using ranges filter
```suggestion
    for (Polygons& tower_roof: tower_roofs | ranges::views::filter([](const auto& poly){ return ! poly.empty(); }))
    {
```

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1824/commits/10194058e601e86d28c240db8db59691f0f1674c